build/tasks/compare_size.mjs

```diff
@@ -0,0 +1,132 @@
+import chalk from "chalk";
+import fs from "fs";
+import { promisify } from "util";
+import zlib from "node:zlib";
+import { exec as nodeExec } from "child_process";
+
+const gzip = promisify( zlib.gzip );
+const exec = promisify( nodeExec );
+
+async function getBranchName() {
+	const { stdout } = await exec( "git rev-parse --abbrev-ref HEAD" );
+	return stdout.trim();
+}
+
+async function getCache( loc ) {
+	try {
+		const contents = await fs.promises.readFile( loc, "utf8" );
+		return JSON.parse( contents );
+	} catch ( err ) {
+		return {};
+	}
+}
+
+function saveCache( loc, cache ) {
+	return fs.promises.writeFile( loc, JSON.stringify( cache ) );
+}
+
+async function cleanWorkingDir() {
+	const { stdout } = await exec( "git status --untracked-files=no --porcelain" );
+	return !stdout.trim();
+}
+
+function compareSizes( existing, current, padLength ) {
+	if ( typeof current !== "number" ) {
+		return chalk.grey( `${existing}`.padStart( padLength ) );
+	}
+	const delta = current - existing;
+	if ( delta > 0 ) {
+		return chalk.red( `+${delta}`.padStart( padLength ) );
+	}
+	return chalk.green( `${delta}`.padStart( padLength ) );
+}
+
+async function compareSize( { cache = ".sizecache.json", files } = {} ) {
+	if ( !files || !files.length ) {
+		throw new Error( "No files specified" );
+	}
+
+	const branch = await getBranchName();
+	const sizeCache = await getCache( cache );
+
+	let rawPadLength = 0;
+	let gzPadLength = 0;
+	const results = await Promise.all(
+		files.map( async function( filename ) {
+			const contents = await fs.promises.readFile( filename, "utf8" );
+			const size = Buffer.byteLength( contents, "utf8" );
+			const gzippedSize = ( await gzip( contents ) ).length;
```

Comment:
I like that we're using Node's native zlib here. I did try that in the past but got discouraged by different numbers reported than from our previous Grunt task. But you're right we should care more about relative comparisons than keeping extra low absolute numbers.